### DISCUSSION TITLE
How to disable rust-analyzer from running on its own?

### DISCUSSION THREAD

**Discussion by yanshay** [LazyVim/LazyVim#1983] [Q&A] [answered]
Rust-analyzer keeps running on my projects w/o being asked or without even any change in any file.
I sometime have even 10 projects open in different nvim sessions, and I'm not using either of them and then I see several rust-analyzer processes running and my machine chokes. in addition to that I also see my nvim process takes more CPU and that I think happens only with rust projects.
I couldn't find how to not trigger rust-analyzer w/o being asked. Is there a way?

**Comment by jyuan0** [comment]
Maybe pass `autostart = false` for `rust_analyzer` in a nvim-lspconfig config?

**Reply by yanshay** [reply]
I think this prevents `rust_analayzer` from starting which may be better for me, at least on some heavy project. I don't think it will change the behavior of running on its own.
It could be an option though to start/stop LSP when I stop working on a project, maybe try to automate it so inactive editors disable lsp automatically until back in use.

**Reply by dpetka2001** [reply, marked as answer]
I saw this reddit [post](https://www.reddit.com/r/neovim/comments/17omj0j/garbagedaynvim_lsp_garbage_collector_to_free_ram/) a while ago. Haven't tried it, but maybe you would like to test it out to see if it helps you with your problem.

**Reply by yanshay** [reply]
Thanks, looks interesting, will check it out.

**Reply by rendivs925** [reply]
> I saw this reddit [post](https://www.reddit.com/r/neovim/comments/17omj0j/garbagedaynvim_lsp_garbage_collector_to_free_ram/) a while ago. Haven't tried it, but maybe you would like to test it out to see if it helps you with your problem.

thanks, it works :)
